### PR TITLE
Add support for Sylius plugins

### PR DESCRIPTION
**Pull request by dunglas**
They use a special convention: `Bundle` is replaced by `Plugin`.

### CODE REVIEW

**Comment by lchrusciel** [comment]
Thanks for the support for our convention :) We have this already done in our skeleton so that all new plugins will support flex 🎉 
 https://github.com/Sylius/PluginSkeleton/blob/master/src/AcmeSyliusExamplePlugin.php

**Comment by pjedrzejewski** [comment]
Thank you @dunglas! :)

**Comment by OskarStark** [comment]
@dunglas would be great to get this feature 👍

File: src/SymfonyBundle.php [hunk 1]
```diff
@@ -71,14 +71,16 @@ private function extractClassNames(string $namespace): array
         $namespace = trim($namespace, '\\');
         $class = $namespace.'\\';
         $parts = explode('\\', $namespace);
-        $suffix = $parts[count($parts) - 1];
-        if ('Bundle' !== substr($suffix, -6)) {
+        $suffix = $parts[\count($parts) - 1];
+        $endOfWord = substr($suffix, -6);
+        if ('Bundle' !== $endOfWord && 'Plugin' !== $endOfWord) {
```
[the remark this answers is not in the document]
Comment (dunglas) [reply src/SymfonyBundle.php]: done

**Comment by francoispluchino** [comment]
See Symfony Flex extend to other projects to facilitate the configuration is something very good, and I thank the Symfony team for that, because I see that we are moving in the right direction. However, you must not perceive my message as negative, on the contrary, but I think this feature should not be natively integrated with Flex.

As @nicolas-grekas has very well recalled in his [comment](https://github.com/symfony/flex/pull/174#issuecomment-334525569), it is best to avoid adding specific code that we will have to maintain. Given that this is a feature specific to Sylius, this should be integrated only in the Sylius project.

Obviously, in the state, it is not possible to do it, however, I think it would be interesting to reconsider the proposition #173 (and the PR #174) which would allow the team in charge of Sylius to create a plugin for Flex realizing what they need for their project.

In this same comment dated Oct. 5, 2017, @nicolas-grekas indicated:

> Personally, I would thus respectfully recommend to not merge this PR or any similar approach in the short/mid term (nobody knows about the future :)).

In 2 years, the interfaces of the main components of Flex have not changed, and the plugin can now be considered as stabilized, given that it is in version 1.2.3 with 103 deployed versions, without changing the interfaces and private properties needed to extend Flex.

Of course and as mentioned in this [comment](https://github.com/symfony/flex/issues/206#issuecomment-356740101), the PR #174 must be updated, but I think it is now reasonable and interesting to reopen the discussion of the relevance and utility to extend Flex, thus avoiding specific cases like this.

**Comment by dunglas** [comment]
It could be a middle or long term solution. But in the meantime, Sylius is an important part of the Symfony ecosystem, and this very specific conventions is used in the wild for years. While this PR isn't merged, it makes the install or Sylius plugins with Flex (installed by default) bad.

I suggest to:

1. merge this PR as is
2. if #174 get merged, move Sylius support in a separate plugin at this point

**Comment by Pierstoval** [comment]
Why not fork Flex and publish a "SyliusFlex" package as a plugin, and only implement the necessary configurators to make a Sylius plugin be added to any Symfony-based project?.

[HarmonyCMS](https://github.com/harmonycms/flex) did that for their own business.

**Comment by fabpot** [comment]
Because forking Flex should be avoided. It fragments the community for no good reasons.

**Comment by fabpot** [comment]
Thank you @dunglas.

**Comment by francoispluchino** [comment]
@Pierstoval Yes, I also did the same for my tools, however, I'll go back to an old comment that indicated the direction the community will take if there was no possibility to extend Flex, ie, create forks, and reinvent the wheel several times...

**Comment by Pierstoval** [comment]
@fabpot I was proposing a Flex fork only "in the meantime" before #174 can be merged (or discussed, before being merged or refactored)

But what you say here is very important and I cannot agree more on this:

> Because forking Flex should be avoided. It fragments the community for no good reasons.

**Comment by francoispluchino** [comment]
I think we all agree on this, just as we all know the Symfony project capability to give the right directions, and so, I think it would be really interesting if Flex could be extended.

**Pull request merged**